homeassistant/components/idasen_desk/desk.py

```diff
@@ -0,0 +1,151 @@
+"""The IKEA Idasen Desk integration."""
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Callable
+import logging
+
+from attr import dataclass
+from bleak import BleakClient
+from idasen import IdasenDesk
+
+from homeassistant.components import bluetooth
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class Desk:
+    """Wrapper around the IdasenDesk from the idasen library."""
```

Comment:
This should move into a PyPI package

Reply:
This is mostly just "converting" from the python lib to HA (for example, the height from meters to percent) and handling callbacks. Is this really required to be in a separate lib, since the main code is already on another lib?

Reply:
This is mostly just "converting" from the python lib to HA (for example, the height from meters to percent) and handling callbacks. Is this really required to be in a separate lib, since the main code is already on another lib?

Reply:
The smaller the PR the faster you'll get this merged.

Reply:
I will create the `idasen-ha` lib and update this to use it instead.